src/server/server_family.cc

```diff
@@ -1764,22 +1771,29 @@ void ServerFamily::Info(CmdArgList args, ConnectionContext* cntx) {
   }
 
   if (should_enter("PERSISTENCE", true)) {
-    decltype(last_save_info_) save_info;
-    {
-      lock_guard lk(save_mu_);
-      save_info = last_save_info_;
-    }
-    // when when last save
-    append("last_save", save_info->save_time);
-    append("last_save_duration_sec", save_info->duration_sec);
-    append("last_save_file", save_info->file_name);
+    auto save_info = GetLastSaveInfo();
+
+    // when last success save
+    append("last_success_save", save_info.save_time);
+    append("last_saved_file", save_info.file_name);
+    append("last_success_save_duration_sec", save_info.success_duration_sec);
+
     size_t is_loading = service_.GetGlobalState() == GlobalState::LOADING;
     append("loading", is_loading);
+    append("saving", IsSaving());
+    auto curent_durration_sec =
+        start_save_time_ ? (absl::Now() - *start_save_time_) / absl::Seconds(1) : 0;
```

Comment:
the is saving and start_save_time_ setting is not atomic. So to prevent rare cases where dragonfly users ask us why start_save_time_ is not 0 and IsSaving is false I would put this inside 
if(IsSaving()) {
append("saving", true);
append("current_save_duration_sec", curent_durration_sec);

}